atlp-rwanda/Dominators-Phantom-BE
# Bg-fix-update-profile-issues

### What does this PR do?
- Changed PUT method to PATCH method (User might want to update some profile info, not all).
- User profile is created when user account is created/registered by admin. (basics info initially: email, firstName, lastName, and role. Other fields are empty for now).
- User profile is updated, not created when the end point is called.
- Changed Profile.create() method to Profile.update() in updateProfile() function.
- Added `req.file` condition to check if the user added a file so that the profile would be changed by calling `await fileUpload(req)` method.
- Solved an error `req.file.path` path not found.

### Description of Task to be completed?

### How should this be manually tested?
- After pulling this branch `bg-fix-profile-issues` locally and checkout in it. Do `npm i` to get all dependencies.
- Run `npm run migrate` to migrate all models in the local database
- Run `npm run seed` to populate the tables with seeders' values
- Run `npm run dev` to start the server manually
- Use postman to test the endpoints: 
1. First login as an Admin and create new user POST `api/v1/users/login` and  POST `api/v1/users/register` respectively.
2. To check if user account has been created successfully, do GET `api/v1/users/:id` and  to check his profile info do GET `api/v1/users/:id/profile`.
3. To update/change his/her profile info do PATCH `api/v1/v1/users/:id/update`. N.B: In your req.body you will have informations like this or some:  profilePic, lastName, firstName, nationalId, gender, phone, category, bio, province, district, sector, cell or village.
4. Successfully update will show the message: `User Updated Successfully` otherwise will prompt unsuccessfully message.

### Any background context you want to provide?
When we request for the endpoint, the updateProfile function was creating new Profile, instead of updating the existing one. This issue may cause one user to have more than one row in the profile table. 
### What are the relevant pivotal tracker stories?

### Screenshots (if appropriate)

### Questions:
